[API] : Limited user cannot retrieve data from User API

Reported via Support: [102882](https://dotcms.zendesk.com/agent/tickets/102882)
Reproducible as of dotCMS 5.2.8, and latest master.

**Describe the bug**
There are several methods in the `com.dotmarketing.business.UserAPI` class that fail to return data for limited users. For example, the following method:
https://github.com/dotCMS/core/blob/f797f56fd239a15a11e1fff08a7f2e9c667238ac/dotCMS/src/main/java/com/dotmarketing/business/UserAPI.java#L67
Fails to return the expected information when a non-admin user is passed down. A stack trace similar to the following is logged:
```
Caused by: com.dotmarketing.exception.DotSecurityException: The User being passed in doesn't have permission to requested User
	at com.dotmarketing.business.UserAPIImpl.loadByUserByEmail_aroundBody4(UserAPIImpl.java:124) ~[dotcms_5.2.8.1_cc873df.jar:?]
	at com.dotmarketing.business.UserAPIImpl$AjcClosure5.run(UserAPIImpl.java:1) ~[dotcms_5.2.8.1_cc873df.jar:?]
	at org.aspectj.runtime.reflect.JoinPointImpl.proceed(JoinPointImpl.java:149) ~[aspectjrt-1.8.10.jar:?]
	at com.dotcms.aspects.aspectj.AspectJDelegateMethodInvocation.proceed(AspectJDelegateMethodInvocation.java:42) ~[dotcms_5.2.8.1_cc873df.jar:?]
	at com.dotcms.aspects.interceptors.CloseDBIfOpenedMethodInterceptor.invoke(CloseDBIfOpenedMethodInterceptor.java:29) ~[dotcms_5.2.8.1_cc873df.jar:?]
	at com.dotcms.aspects.aspectj.CloseDBIfOpenedAspect.invoke(CloseDBIfOpenedAspect.java:41) ~[dotcms_5.2.8.1_cc873df.jar:?]
	at com.dotmarketing.business.UserAPIImpl.loadByUserByEmail(UserAPIImpl.java:110) ~[dotcms_5.2.8.1_cc873df.jar:?]
```

**To Reproduce**
Steps to reproduce the behavior:
1. Start up a dotCMS instance
2. Create a JSP file with the following code, and put it under `/webapps/ROOT/` so that it can be accessed throught the browser as http://localhost:8080/test.jsp :
```java
<%@ page import="com.dotmarketing.business.UserAPI" %>
<%@ page import="com.dotmarketing.business.APILocator" %>
<%@ page import="com.liferay.portal.model.User" %>
<%@ page contentType="text/html;charset=UTF-8" language="java" %>
<html>
<head>
    <title>Test JSP</title>
</head>
<body>
<%
    final UserAPI userAPI = APILocator.getUserAPI();
    final User limitedUser = userAPI.loadUserById("dotcms.org.2793"); // Daniel Publisher
    //User ownUser = userAPI.loadByUserByEmail("daniel@dotcms.com", limitedUser, false);
    User ownUser = userAPI.loadByUserByEmail("daniel@dotcms.com", APILocator.systemUser(), false);
%>
	<p>
		User Data = <%= ownUser.toString() %>
	</p>
</body>
</html>
```
3. User data will be displayed on the page. Now, uncomment line 13, and comment out line 14.
4. You'll see a Java stack trace in the log.

**Expected behavior**
A limited user should able to access and change user data. This should not be limited to admin users only.

**Additional context**
For the specific case of the `loadByUserByEmail` method -- which is similar to other restricted methods in the same API -- the code that is throwing the exception is the following:
```java
if(permissionAPI.doesUserHavePermission(userProxyAPI.getUserProxy(u,APILocator.getUserAPI().getSystemUser(), false), PermissionAPI.PERMISSION_READ, user, respectFrontEndRoles)){
        return u;
}else{
        throw new DotSecurityException("The User being passed in doesn't have permission to requested User");
}
```

**Official Workaround**
Call the failing method using the System User instead of a limited User, like this:
```java
APILocator.getUserAPI().getSystemUser()
```

--- a/dotCMS/src/integration-test/java/com/dotmarketing/business/UserAPITest.java
+++ b/dotCMS/src/integration-test/java/com/dotmarketing/business/UserAPITest.java
@@ -1268,4 +1268,74 @@ public class UserAPITest extends IntegrationTestBase {
         assertEquals("MyFacebookId", savedUser.getAdditionalInfo().get("facebookId"));
         assertEquals("MyTwitterId", savedUser.getAdditionalInfo().get("twitterId"));
     }
+
+	/**
+	 * Method to test: {@link UserAPI#loadByUserByEmail}
+	 * Given Scenario: Given limited user
+	 * ExpectedResult: Limited user should be able to load self-user
+	 *
+	 */
+	@Test
+	public void test_loadByUserByEmail_selfUser() throws DotDataException, DotSecurityException {
+
+		final User newUser = new UserDataGen().firstName("limitedUser" + System.currentTimeMillis())
+			.nextPersisted();
+
+		userAPI.save(newUser, systemUser, false);
+
+		final User ownUser = userAPI.loadByUserByEmail(newUser.getEmailAddress(),
+				newUser, false);
+
+		assertNotNull(ownUser.getAdditionalInfo());
+		assertEquals(newUser.getUserId(), ownUser.getUserId());
+	}
+
+	/**
+	 * Method to test: {@link UserAPI#loadUserById(String, User, boolean)}
+	 * Given Scenario: Given limited user
+	 * ExpectedResult: Limited user should be able to load self-user
+	 *
+	 */
+	@Test
+	public void test_loadByUserById_selfUser() throws DotDataException, DotSecurityException {
+
+		final User newUser = new UserDataGen().firstName("limitedUser" + System.currentTimeMillis())
+				.nextPersisted();
+
+		userAPI.save(newUser, systemUser, false);
+
+		final User ownUser = userAPI.loadUserById(newUser.getUserId(),
+				newUser, false);
+
+		assertEquals(newUser.getUserId(), ownUser.getUserId());
+	}
+
+	/**
+	 * Method to test: {@link UserAPI#save(User, User, boolean)}
+	 * Given Scenario: Given limited user
+	 * ExpectedResult: Limited user should be able to save self-user
+	 *
+	 */
+	@Test
+	public void test_save_selfUser() throws DotDataException, DotSecurityException {
+
+		final User newUser = new UserDataGen().firstName("limitedUser" + System.currentTimeMillis())
+				.nextPersisted();
+
+		final String userId = newUser.getUserId();
+
+		userAPI.save(newUser, systemUser, false);
+
+		final User ownUser = userAPI.loadUserById(userId,
+				APILocator.systemUser(), false);
+
+		// let's modify the user and save it using the self-user
+		ownUser.setFirstName("modifiedLimitedUser"+ + System.currentTimeMillis());
+		userAPI.save(ownUser, ownUser, false);
+
+		final User reloadedUser = userAPI.loadUserById(userId,
+				APILocator.systemUser(), false);
+
+		assertTrue(reloadedUser.getFirstName().contains("modifiedLimitedUser"));
+	}
 }

--- a/dotCMS/src/main/java/com/dotmarketing/business/PermissionBitAPIImpl.java
+++ b/dotCMS/src/main/java/com/dotmarketing/business/PermissionBitAPIImpl.java
@@ -257,7 +257,7 @@ public class PermissionBitAPIImpl implements PermissionAPI {
 
 	private boolean userPermissions(final UserProxy userProxy, final User userIn) {
 	    
-	    if(userProxy.getPermissionId().equals(userIn.getUserId())) {
+	    if(userProxy.getPermissionId().equals("user:"+userIn.getUserId())) {
 	        return true;
 	    }
 	    return Try.of(()-> APILocator.getLayoutAPI().doesUserHaveAccessToPortlet("user", userIn)).getOrElse(false);

